WorkloadManagementSystem/Executor/InputData.py

```diff
@@ -372,3 +401,44 @@ def __getSiteCandidates(self, okReplicas, vo):
       sitesData[siteName]['disk'] = len(sitesData[siteName]['disk'])
       sitesData[siteName]['tape'] = len(sitesData[siteName]['tape'])
     return S_OK(sitesData)
+
+  @executeWithUserProxy
+  def _resolveInputSandbox(self, jobState, inputSandbox):
+    """ This method checks the file catalog for replica information.
+
+        :param JobState jobState: JobState object
+        :param list inputSandbox: list of LFNs for the input sandbox
+
+        :returns: S_OK/S_ERROR structure with resolved input data info
+    """
+
+    result = jobState.getManifest()
+    if not result['OK']:
+      self.jobLog.error("Failed to get job manifest", result['Message'])
+      return result
+    manifest = result['Value']
+    vo = manifest.getOption('VirtualOrganization')
+    startTime = time.time()
+    dm = self.__getDataManager(vo)
+    if dm is None:
+      return S_ERROR('Failed to instantiate DataManager for vo %s' % vo)
+
+    # This will return already active replicas, excluding banned SEs, and
+    # removing tape replicas if there are disk replicas
+
+    result = dm.getReplicasForJobs(inputSandbox)
+    self.jobLog.verbose('Catalog replicas lookup time', '%.2f seconds ' % (time.time() - startTime))
+    if not result['OK']:
+      self.log.warn(result['Message'])
+      return result
+
+    isDict = result['Value']
+
+    self.jobLog.verbose("REPLICA DICT", isDict)
+
+    result = self.__checkReplicas(isDict, vo)
+
+    if not result['OK']:
+      self.jobLog.error("Failed to check replicas", result['Message'])
+
```

Comment:
This whole method is largely a copy of the first lines of the `_resolveInputData` method. Would you like to unify the 2?

Reply:
I tried at first, but it makes the code even more intertwined, so I decided to split it this way